web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -0,0 +1,354 @@
+      'creation date' => 'field_ad_creation_date',
+      'description' => 'field_ad_description',
+      'file' => 'field_ad_file',
+      'file size' => 'field_ad_file_size',
+      'format' => 'field_ad_format',
+      'licence' => 'field_ad_licence',
+      'modification date' => 'field_ad_modification_date',
+      'representation technique' => 'field_ad_repr_technique',
+      'status' => 'field_ad_status',
+    ];
+  }
+
+  /**
+   * Creates an asset distribution with data provided in a table.
+   *
+   * Table format:
+   * | title             | Sample distribution                          |
+   * | uri               | http://joinup.eu/distribution/foobar         |
+   * | description       | A sample distribution                        |
+   * | access url        | http://internal-or-external.path/to/file.zip |
+   * | file              | test.zip                                     |
+   *
+   * Fields title and description are required.
+   *
+   * @param TableNode $asset_distribution_table
+   *   The asset distribution data.
+   *
+   * @throws \Exception
+   *   Thrown when a column name is incorrect.
+   *
+   * @Given (the following )asset distribution:
+   */
+  public function givenAssetDistribution(TableNode $asset_distribution_table) {
+    $aliases = self::assetDistributionFieldAliases();
+
+    $values = [];
+    // Replace the column aliases with the actual field names.
+    foreach ($asset_distribution_table->getRowsHash() as $key => $value) {
+      if (array_key_exists($key, $aliases)) {
+        $values[$aliases[$key]] = $value;
+      }
+      else {
+        throw new \Exception("Unknown column '$key' in solution table.");
+      }
+    };
+    $this->createAssetDistribution($values);
+  }
+
+  /**
+   * Creates a asset distribution from the given property and field data.
+   *
+   * @param array $values
+   *   An optional associative array of values, keyed by property name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   A new asset distribution entity.
+   *
+   * @throws \Exception
+   *   Thrown when a given file is not found.
+   */
+  public function createAssetDistribution(array $values) {
+    // Add image.
+    if (!empty($values['field_ad_file'])) {
+      $filename = $values['field_ad_file'];
+      $files_path = $this->getMinkParameter('files_path');
+      $path = rtrim(realpath($files_path), DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $filename;
+      if (!is_file($path)) {
+        throw new \Exception("File '$filename' was not found in file path '$files_path'.");
+      }
+      // Copy the file into the public files folder and turn it into a File
+      // entity before linking it to the collection.
+      $uri = 'public://' . $filename;
+      $destination = file_unmanaged_copy($path, $uri);
+      $file = File::create(['uri' => $destination]);
+      $file->save();
+      $values['field_ad_file'] = [$file->id()];
+
+      $this->files[$file->id()] = $file;
+    }
```

Comment:
This code is duplicated from the `CollectionSubContext`. Let's create a trait that contains code related to working with files, so that we can reuse it.